numba/cuda/printimpl.py

```diff
@@ -69,18 +53,26 @@ def print_varargs(context, builder, sig, args):
     It dispatches to the appropriate 'print' implementations above
     depending on the detected real types in the signature."""
 
-    mod = builder.module
-    vprint = nvvmutils.declare_vprint(mod)
-    sep = context.insert_string_const_addrspace(builder, " ")
-    eol = context.insert_string_const_addrspace(builder, "\n")
+    vprint = nvvmutils.declare_vprint(builder.module)
+
+    formats = []
+    values = []
 
     for i, (argtype, argval) in enumerate(zip(sig.args, args)):
+        # XXX The return type is a lie: "print_item" doesn't return
```

Comment:
I thought about that. The downside is that it makes it harder for people to extend print() for other types. I wonder if there is such a need, though. What do you think?